File: Sharebook/Client/src/app/core/Public/Parameters/RateLimitsClient/WaitForCredentialsRateLimitParameters.ts
import {ITwitterExecutionContext} from "../../../Core/Client/TwitterExecutionContext";
import {RateLimitsSource} from "../HelpClient/GetRateLimitsParameters";
import {IReadOnlyTwitterCredentials} from "../../../Core/Models/Authentication/ReadOnlyTwitterCredentials";

export interface IWaitForCredentialsRateLimitParameters {
  url: string;
  credentials: IReadOnlyTwitterCredentials;
  executionContext: ITwitterExecutionContext;
  from: RateLimitsSource;
}

export class WaitForCredentialsRateLimitParameters implements IWaitForCredentialsRateLimitParameters {
  constructor(url: string) {
    this.url = url;
  }

  public url: string;
  public credentials: IReadOnlyTwitterCredentials;
  public executionContext: ITwitterExecutionContext;
  public from: RateLimitsSource;
}

File: Sharebook/Server/src/Services/Identity/Sharebook.Identity.API/Application/Parameters/RateLimitsClient/WaitForCredentialsRateLimitParameters.cs
namespace Sharebook.Identity.API.Application.Parameters.RateLimitsClient
{
    using Sharebook.Common.Public.Parameters.HelpClient;
    using Sharebook.Identity.API.Auth.Authentication;
    using Sharebook.Identity.API.Client;

    public interface IWaitForCredentialsRateLimitParameters
    {
        string Url { get; set; }

        IReadOnlyTwitterCredentials Credentials { get; set; }

        ITwitterExecutionContext ExecutionContext { get; set; }

        RateLimitsSource From { get; set; }
    }

    public class WaitForCredentialsRateLimitParameters : IWaitForCredentialsRateLimitParameters
    {
        public WaitForCredentialsRateLimitParameters(string url)
        {
            Url = url;
        }

        public string Url { get; set; }

        public IReadOnlyTwitterCredentials Credentials { get; set; }

        public ITwitterExecutionContext ExecutionContext { get; set; }

        public RateLimitsSource From { get; set; }
    }
}
